learn/validator-trust-score.md

```diff
@@ -0,0 +1,227 @@
+
+</Callout>
+
+<details>
+
+<summary children:m-0>
+
+Details on how to calculate the $r_i$
+
+</summary>
+
+The number of blocks that the validators produced can be fetched from the blockchain via the inherents of a batch. We will refer to these value as $b_j$, where $j \in [0, B]$. $B$ is the number of batches in an epoch.
+
+Figuring out how many blocks a validator is likely to produce is a bit trickier. In each voting block we can see the slot distribution for the next epoch. Basically, there will be a value from 1 to 512 for each active validator. A value of 1 means that the validator is likely to make 1 out of 512 blocks, while a value of 512 means that the validator is likely to make all the blocks. We need to convert this number into a probability, a value between 0 and 1.
+
+Here, $s_j$ is the slot distribution value for the batch $j$. This value tells us the probability that a validator will produce a block in that batch.
+
+</details>
+
+$$
+r_i = \frac{\sum_{j=0}^{B} b_j}{\sum_{j=0}^{B} s_i}
+$$
+
+where $l_0$ is the liveness value of the most recent epoch and $l_{m-1}$ is the liveness of the oldest epoch.
+
+To combine all the liveness scores into a single value, we do a moving average, where more recent epochs have higher weights than older ones.
+
+$$
+
+\bar{L} = \frac{\sum_{i=0}^{m-1} \left( 1-a\frac{i}{m-1} \right) l_i}{\sum_{i=0}^{m-1} \left( 1-a\frac{i}{m-1} \right)}, \quad a = 0.5
+
+$$
+
+Being $a$, the parameter determining how much the observation of the oldest epoch is worth relative to the observation of the newest epoch.
+
+#### Adjusting for High-Reliability Expectations
+
+The previous formula provides a weighted moving average score for server reliability in block production, where a score of $0.9$ indicates a significant downtime of 10%, highlighting recent performance and the need for improved consistency.
+
+As one block is missing, this is a significant shortfall. To better reflect the high standards required, we will plot the value on a circle, having $c$ as the parameter defining the slope of the arc.
+
+$$
+R=-c+1-\sqrt{-\bar{R}^{2}+2c\bar{R}+\left(c-1\right)^{2}}
+$$
+
+$$
+\text{Center of circle at (c,-c+1), where }
+c=-0.16
+$$
+
+<figure>
+
+<iframe src="https://www.desmos.com/calculator/zqemsh7yay?embed" aspect-video frameborder="0" allowfullscreen></iframe>
+
+<figcaption>
+
+Graph of the reliability score adjustment. The x-axis represents the reliability score, and the y-axis represents the adjusted reliability score.
+
+</figcaption>
+
+</figure>
+
+## Suggestions & Feedback
+
+Like everything in Nimiq, this algorithm is designed for people. We know it is not perfect and we may have missed some details. If you have any feedback or suggestions that you think will improve the algorithm, please feel free to contact us by [opening an issue](https://github.com/nimiq/developer-center/issues?q=is:issue+is:open+sort:updated-desc) or in the [Telegram group](https://t.me/joinchat/AAAAAEJW-ozFwo7Er9jpHw).
```

Comment:
I will add a comment here stating that the curves and constants presented in this document may change any time in the future without previous notice

Reply:
![image](https://github.com/nimiq/developer-center/assets/22072217/e811bae2-eddf-4b75-bf4d-e8764a92b4f8)
I added a callout before going into detail instead